File: Data Structures & Algorithms/remove-element/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} val
     * @return {number}
     */
    removeElement(nums, val) {
        let i = 0;
        let j = 0;
        let len = nums.length;

        while (i < len && j < len) {
            if(nums[i] === val) {
                while (j<len) {
                    if(nums[j] !== val) break;
                    j++;
                }
                if(j>=len) break;

                let temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                i++;
            } else {
                i++;
                j++;
            }
        }

        return i;
    }
}

File: Data Structures & Algorithms/first-missing-positive/submission-3.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int n=nums.size();

       unordered_map<int,int>mp;


        int positivecount=0;


        for(auto it:nums){

            if(it>0){
                mp[it]++;
                positivecount++;
            }
        }


  for(int i=1;i<=positivecount+1;i++){


    if(!mp[i])return i;
  }







    }
};